orchestration/dagster_orchestration/hca_orchestration/solids/load_hca/load_data_files.py

```diff
@@ -1,12 +1,236 @@
+        file_load_table_name: str
+) -> bigquery.ExtractJob:
+    # dump the contents of our file load requests table to a file in GCS suitable for use as a
+    # control file for jade bulk data ingest
+    out_prefix = _gs_path_from_bucket_prefix(scratch_config.scratch_bucket_name, scratch_config.scratch_prefix_name)
+    out_path = f"{out_prefix}/data-transfer-requests-deduped/*"
+
+    job_config = bigquery.job.ExtractJobConfig()
+    job_config.destination_format = bigquery.DestinationFormat.NEWLINE_DELIMITED_JSON
+    extract_job: bigquery.ExtractJob = bigquery_client.extract_table(
+        f"{scratch_dataset_name}.{file_load_table_name}",
+        destination_uris=[
+            f"{out_path}"
+        ],
+        job_config=job_config,
+        location="US",
+        project=scratch_config.scratch_bq_project
+    )
+    return extract_job.result()
+
 
-from hca_orchestration.solids.load_hca.stage_data import HcaStagingDatasetName
+@solid(
+    required_resource_keys={"data_repo_client", "scratch_config", "load_tag", "target_hca_dataset", "storage_client"},
+    input_defs=[InputDefinition("control_file_path", str)]
+)
+def run_bulk_file_ingest(context: AbstractComputeExecutionContext, control_file_path: str) -> JobId:
+    """
+    Submits the given control for ingestion to TDR
+    :param context: Dagster solid context
+    :param control_file_path: Path to the control file for ingest
+    :return: Jade Job ID
+    """
+    profile_id = context.resources.target_hca_dataset.billing_profile_id
+    dataset_id = context.resources.target_hca_dataset.dataset_id
+    scratch_bucket_name = context.resources.scratch_config.scratch_bucket_name
+
+    payload = {
+        "profileId": profile_id,
+        "loadControlFile": f"gs://{scratch_bucket_name}/{control_file_path}",
+        "loadTag": context.resources.load_tag,
+        "maxFailedFileLoads": 0
+    }
+    context.log.info(f'Bulk file ingest payload = {payload}')
+
+    data_repo_client = context.resources.data_repo_client
+    job_response: JobModel = data_repo_client.bulk_file_load(
+        dataset_id,
+        bulk_file_load=payload
+    )
+    context.log.info(f"bulk file ingest job id = {job_response.id}")
+    result: JobId = JobId(job_response.id)
+
+    return result
 
 
 @solid(
-    input_defs=[InputDefinition("staging_dataset_name", HcaStagingDatasetName)]
+    required_resource_keys={"data_repo_client"},
+    config_schema={
+        'max_wait_time_seconds': Int,
+        'poll_interval_seconds': Int,
+    }
+)
+def _base_check_bulk_file_ingest_job_result(context: AbstractComputeExecutionContext, job_id: JobId) -> Nothing:
+    # we need to poll on the endpoint as a workaround for a race condition in TDR (DR-1791)
+    job_results = api_request_with_retry(
+        context.resources.data_repo_client.retrieve_job_result,
+        context.solid_config['max_wait_time_seconds'],
+        context.solid_config['poll_interval_seconds'],
+        {500},
```

Comment:
i'm wondering if we can save ourselves some future pain here by marking other common transient 5xx errors (i.e. 502-504) for retry as well